src/enclave/enclave_oe.c

```diff
@@ -61,40 +178,71 @@ void sgxlkl_ethread_init(void)
     sched_tcb->schedctx = (struct schedctx*)((char*)tls_page + tls_offset);
 
     /* Wait until libc has been initialized */
-    while (__libc_state != libc_initialized)
+    while (sgxlkl_enclave_state.libc_state != libc_initialized)
     {
         a_spin();
     }
 
     /* Initialization completed, now run the scheduler */
     __init_tls();
-    _lthread_sched_init(sgxlkl_enclave->stacksize);
+    _lthread_sched_init(sgxlkl_enclave_state.config->stacksize);
     lthread_run();
 
     return;
 }
 
-int sgxlkl_enclave_init(const sgxlkl_config_t* config_on_host)
+static int _read_eeid_config()
 {
-    SGXLKL_ASSERT(config_on_host);
+    const oe_eeid_t* eeid = (oe_eeid_t*)__oe_get_eeid();
+    const char* config_json = (const char*)eeid->data;
+    sgxlkl_enclave_state.libc_state = libc_not_started;
 
+    sgxlkl_enclave_config_t* cfg = malloc(sizeof(sgxlkl_enclave_config_t));
+    sgxlkl_read_enclave_config(config_json, cfg, true);
+    sgxlkl_enclave_state.config = cfg;
+    return 0;
+}
+
+static int _copy_shared_memory(const sgxlkl_shared_memory_t* shm)
+{
+    // Copy shared memory. Deep copy so the host can't change it?
+    memcpy(
+        &sgxlkl_enclave_state.shared_memory,
+        shm,
+        sizeof(sgxlkl_shared_memory_t));
+
+    // This will be removed once shared memory and config have been
+    // separated fully.
```

Comment:
Is there an open GitHub issue for this? Can you explain this better?

Reply:
No and no :-) There's no point in opening an github issue at this point. There are a number of old code lines that refer to the global variable called `sgxlkl_enclave`, which now points to the new config structure, but has the same member names, so I didn't have to change all occurrences right away. I think they are all in our `src/enclave` and `src/lkl` now, so we could refactor them to point to the new config and get rid of `sgxlkl_enclave` if we want to.

Reply:
Is the refactoring you mention part of this PR? If not, we need a better way to track this as a task.

Reply:
There were fewer references left over than I thought, so this is done too.